File: rules/gitlab/scala/file/test-FileUploadFileName.scala
// License: LGPL-3.0 License (c) find-sec-bugs
package file

import org.apache.commons.fileupload.FileItem
import org.apache.commons.fileupload.FileUploadException
import org.apache.commons.fileupload.disk.DiskFileItemFactory
import org.apache.commons.fileupload.servlet.ServletFileUpload
import javax.servlet.http.HttpServletRequest
import java.util._
import scala.jdk.CollectionConverters._

class FileUploadFileName {
  @throws[FileUploadException]
  def handleFileCommon(req: HttpServletRequest): Unit = {
    val upload = new ServletFileUpload(new DiskFileItemFactory())
    val fileItems = upload.parseRequest(req)
    for (item <- fileItems.asScala) {
      println("Saving " + item.getName() + "...")
    }
  }
}

File: rules/gitlab/java/file/test-FileUploadFileName.java
// License: LGPL-3.0 License (c) find-sec-bugs
package file;

import org.apache.commons.fileupload.FileItem;
import org.apache.commons.fileupload.FileUploadException;
import org.apache.commons.fileupload.disk.DiskFileItemFactory;
import org.apache.commons.fileupload.servlet.ServletFileUpload;
import javax.servlet.http.HttpServletRequest;
import java.util.List;

public class FileUploadFileName {

    public void handleFileCommon(HttpServletRequest req) throws FileUploadException {
        ServletFileUpload upload = new ServletFileUpload(new DiskFileItemFactory());
        List<FileItem> fileItems = upload.parseRequest(req);

        for (FileItem item : fileItems) {
            System.out.println("Saving " + item.getName() + "...");
        }
    }

}
